src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesPrisonOffenderEventListener.kt

```diff
@@ -33,33 +34,50 @@ class IncentivesPrisonOffenderEventListener(
-            sqsMessage.Message
-          )
-          log.debug("received IEP_UPSERTED Offender event for $offenderIdDisplay bookingId $bookingId and seq $iepSeq with auditModuleName $auditModuleName")
-          if (shouldSynchronise(auditModuleName)) {
-            incentivesSynchronisationService.synchroniseIncentive(objectMapper.readValue(sqsMessage.Message))
+        "Notification" -> {
+          val eventType = sqsMessage.MessageAttributes!!.eventType.Value
+          if (eventFeatureSwitch.isEnabled(eventType)) when (eventType) {
+            "IEP_UPSERTED" -> {
+              val (offenderIdDisplay, bookingId, iepSeq, auditModuleName) = objectMapper.readValue<IncentiveUpsertedOffenderEvent>(
+                sqsMessage.Message
+              )
+              log.debug("received IEP_UPSERTED Offender event for $offenderIdDisplay bookingId $bookingId and seq $iepSeq with auditModuleName $auditModuleName")
+              if (shouldSynchronise(auditModuleName)) {
+                incentivesSynchronisationService.synchroniseIncentive(objectMapper.readValue(sqsMessage.Message))
+              }
+            }
+
+            "IEP_DELETED" -> {
+              val (offenderIdDisplay, bookingId, iepSeq) = objectMapper.readValue<IncentiveDeletedOffenderEvent>(
+                sqsMessage.Message
+              )
+              log.debug("received IEP_DELETED Offender event for $offenderIdDisplay bookingId $bookingId and seq $iepSeq")
+              incentivesSynchronisationService.synchroniseDeletedIncentive(objectMapper.readValue(sqsMessage.Message))
+            }
+
+            else -> log.info("Received a message I wasn't expecting: {}", eventType)
+          } else {
+            log.warn("Feature switch is disabled for {}", eventType)
           }
         }
 
-        "IEP_DELETED" -> {
-          val (offenderIdDisplay, bookingId, iepSeq) = objectMapper.readValue<IncentiveDeletedOffenderEvent>(sqsMessage.Message)
-          log.debug("received IEP_DELETED Offender event for $offenderIdDisplay bookingId $bookingId and seq $iepSeq")
-          incentivesSynchronisationService.synchroniseDeletedIncentive(objectMapper.readValue(sqsMessage.Message))
-        }
+        IncentiveMessages.SYNCHRONISE_CURRENT_INCENTIVE.name -> incentivesSynchronisationService.handleSynchroniseCurrentIncentiveMessage(
+          sqsMessage.Message.fromJson()
+        )
 
-        else -> log.info("Received a message I wasn't expecting {}", eventType)
-      } else {
-        log.info("Feature switch is disabled for event {}", eventType)
+        IncentiveMessages.RETRY_INCENTIVE_SYNCHRONISATION_MAPPING.name -> incentivesSynchronisationService.retryCreateIncentiveMapping(
+          sqsMessage.Message.fromJson()
+        )
       }
     }.thenAccept { }
   }
 
+  private inline fun <reified T> String.fromJson(): T =
+    objectMapper.readValue(this, object : TypeReference<T>() {})
+
```

Comment:
very minor, you shouldn't the second argument in `readValue`  assuming you import in the single argument Kotlin extension